### ISSUE TITLE
macOS Auto updater using old update

### ISSUE THREAD

**Issue by Timer** [electron-userland/electron-builder#1200]
See log for details, but the auto updater is installing a stale (old) update:

Version 1.0.15 downloads update 1.0.16 but then proceeds to use update 1.0.15 instead of 1.0.16. I've re-opened and re-ran 10x times -- doesn't seem to be a fluke.

```
Josephs-MacBook-Pro:/ joe$ ./Applications/StatsPull.app/Contents/MacOS/StatsPull 
Wait for app ready
Checking for update
Found version 1.0.16 (url: https://s3.amazonaws.com/statspull/statspull-1.0.16-mac.zip)
Downloading update from https://s3.amazonaws.com/statspull/statspull-1.0.16-mac.zip
2017-02-02 02:17:34.230 StatsPull[85494:2941123] Download completed to: file:///Users/joe/Library/Caches/com.statspull.ShipIt/update.ZKND6Jl/statspull-1.0.15-mac.zip
New version 1.0.16 has been downloaded
```

Removing `statspull-1.0.15-mac.zip` from the S3 bucket results in the following error:
```
Josephs-MacBook-Pro:/ joe$ ./Applications/StatsPull.app/Contents/MacOS/StatsPull 
Wait for app ready
Checking for update
Found version 1.0.16 (url: https://s3.amazonaws.com/statspull/statspull-1.0.16-mac.zip)
Downloading update from https://s3.amazonaws.com/statspull/statspull-1.0.16-mac.zip
{ Error: Update download failed. The server sent an invalid response. Try again later.
  preventDefault: [Function: preventDefault],
  sender: 
   AutoUpdater {
     _events: { error: [Function], 'update-downloaded': [Function] },
     _eventsCount: 2 } }
Error: Error: Update download failed. The server sent an invalid response. Try again later.
```

Is the `mac-latest.json` being cached somewhere and telling it to download `statspull-1.0.15-mac.zip`?

----

electron-updater 1.3.2
electron-builder 12.2.2
... just upgraded from 1.3.1 and 12.0.3

**Comment by develar** [comment]
It seems that native Squirrel.Mac caches update json.

You can set `autoUpdater.requestHeaders = {"Cache-Control": "no-cache"}` as workaround.

**Comment by develar** [comment]
Published as 1.4.1

**Comment by Timer** [comment]
Thanks for the quick turnaround!
